src/classes/ADDR_Contact_TDTM.cls

```diff
@@ -135,13 +147,13 @@ public class ADDR_Contact_TDTM extends TDTM_Runnable {
             	//     - if MailingAddress field changes, update/new address object
                 //     - if isAddressOverride changes to false, refill from Default Address
             	
-            	Contact conOld = (Contact)listOld[i];
-            	
             	// new highest priority! if Account changes... 
-                if (conOld != null && con.AccountId != conOld.AccountId) {
+                if (acctChanged) {
                     // if no override, refill from the Default Address
             	    if (!con.is_Address_Override__c && !isContactAddressChanged(con, conOld)) {
-                        listConAddrReset.add(con);
+                        if (acctHasAddress) { // refresh address if the contact does not have one and account does
```

Comment:
if the acct doesn't have an address, I see you correctly avoid adding the contact to the reset address list, but don't we then want to do have a new address object created on the account, just like it was an override, and the contact added to listConCreateAddr?